### ISSUE TITLE
Move all Halibot libraries into a package

### ISSUE THREAD

**Issue by richteer** [Halibot/halibot#9]
Move all of the module classes, subclasses, etc. to a directory labeled halmodule, so that they all exist under one directory. Furthermore, simplifies importing all of them, since they can be imported en masse. Ideally, this should have been done in the first place.

Example:

``` python
import halmodule

if issubclass(halmodule.Module, halmodule.CommandModule):
   celebrate()
```

Directory tree:

```
halmodule/
  __init__.py
  Module.py
  CommandModule.py
```

Also requires renaming halmodule -> Module again, which is a bit of a refactor.

**Comment by richteer** [comment]
Hmmm... turns out this is a little nastier than originally expected. Forgot you still will need to import the class from within the module, so the backend code might look something like:

``` python
from halmodule import Module
from halmodule import CommandModule

m = Module.Module()
n = CommandModule.CommandModule()
```

and the module developer's would be something like:

``` python
from halmodule import Module,CommandModule

class TestModule(Module.Module):
   pass

class QuoteModule(CommandModule.CommandModule):
   pass
```

Still researching possibilties to make this a bit nicer.

**Comment by sjrct** [comment]
I am against the `HalModule` --> `Module` change as it serves to create some measure of ambiguity with the builtin python `module` class.

**Comment by richteer** [comment]
Implemented in PR #27
